Bindle.Tests/Integration.cs

```diff
@@ -1,19 +1,156 @@
+        public IntegrationFixture()
+        {
+            var bindlePath = Environment.GetEnvironmentVariable("BINDLE_SERVER_PATH");
+            if (!string.IsNullOrEmpty(bindlePath))
+            {
+                var fullPath = Path.GetFullPath(Path.Join(bindlePath, "bindle-server"));
+                dataPath = Path.GetFullPath(Path.Join("../../../data"));
+                Console.WriteLine($"Using Bindle Server Path: {fullPath}");
+                Console.WriteLine($"Using Test Data Path: {dataPath}");
+                var psi = new ProcessStartInfo
+                {
+                    FileName = fullPath,
+                    Arguments = $"-i 127.0.0.1:14044 -d {dataPath}",
+                    UseShellExecute = false,
+                    CreateNoWindow = true,
+                    RedirectStandardError = true,
+                    RedirectStandardOutput = true
+                };
+                psi.Environment["RUST_LOG"] = "info";
+                process = new Process
+                {
+                    StartInfo = psi
+                };
+                process.OutputDataReceived += (sender, e) =>
+                {
+                    if (!String.IsNullOrEmpty(e.Data))
+                    {
+                        Console.WriteLine(e.Data);
+                    }
+                };
+                process.ErrorDataReceived += (sender, e) =>
+                {
+                    if (!String.IsNullOrEmpty(e.Data))
+                    {
+                        Console.WriteLine(e.Data);
+                    }
+                };
+                process.Start();
+                process.BeginErrorReadLine();
+                process.BeginOutputReadLine();
+                // TODO: something a bit less crappy that this to wait for server to start:
+                Thread.Sleep(5000);
+                Assert.False(process.HasExited);
+
+                // data/invoices/33d7f76b3077de5b72f50d3bf1997d6571770665732ac83b3bede75cf3c5fb67/invoice.toml gets modified by tests so take a copy so it can be reverted at the end.
+
+                origFilePath = Path.Join(dataPath, "invoices/33d7f76b3077de5b72f50d3bf1997d6571770665732ac83b3bede75cf3c5fb67/invoice.toml");
+                copyFilePath = Path.Join(dataPath, "invoices/33d7f76b3077de5b72f50d3bf1997d6571770665732ac83b3bede75cf3c5fb67/invoice.toml.copy");
+                File.Copy(origFilePath, copyFilePath);
+
+            }
+        }
+
+        public void Dispose()
+        {
+
+            // Stop the bindle server
+            try
+            {
+                process?.Kill();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Failed to stop bindle process:{ex}");
+            }
+
+            // Delete the files and dirs that the tests created.
+
+            var dirs = new string[]
+            {
+                Path.Join(dataPath, "invoices/8fb420e8308462667f8e1585b7e768bf2e391aec9a29d13c968849c99ed5b0bf"),
```

Comment:
Is it possible to reduce maintenance by discovering these directories at run time?  E.g. capture the list of directories before running the first test, then at the end delete any directories that are not in the list?

Reply:
yes that sounds great , I will update it.